docs/src/pages/docs/reference/rag/rerank.mdx

```diff
@@ -0,0 +1,183 @@
+---
+title: "Reference: Rerank | Document Retrieval | RAG | Mastra Docs"
+description: Documentation for the rerank function in Mastra, which provides advanced reranking capabilities for vector search results.
+---
+
+# rerank()
+
+The `rerank()` function provides advanced reranking capabilities for vector search results by combining semantic relevance, vector similarity, and position-based scoring. 
+
+## Usage Example
+
+```typescript
+import { rerank } from "@mastra/rag";
+
+const rerankedResults = await rerank(
+  vectorSearchResults,
+  "How do I deploy to production?",
+  {
+    rerankProvider: "agent",
+    agentModel: {
+      provider: "openai",
+      name: "gpt-4"
+    }
+  },
+  {
+    weights: {
+      semantic: 0.5,
+      vector: 0.3,
+      position: 0.2
+    },
+    topK: 3
+  }
+);
+```
```

Comment:
The parameters signature is:

`rerank( searchResults, queryString, modelConfig, rerankOptions )` 

right? I don't see that statement anywhere